### PR TITLE
admin verification display logic for american indian status self attestation

### PR DESCRIPTION
**Pull request by ATBull81**
# PR Checklist

Please check if your PR fulfills the following requirements:
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bug fixes/features), and they use `let` helpers and `before` blocks.
- [x] For all UI changes, there is Cucumber coverage.
- [x] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, the reasoning is documented in the PR and code.
- [x] Any endpoint modified in the PR only responds to the expected MIME types.
- [x] For all scripts or rake tasks, how to run them is documented in both the PR and the code.
- [x] There are no inline styles added.
- [x] There is no inline JavaScript added.
- [x] There is no hard-coded text added/updated in helpers/views/JavaScript. New/updated translation strings do not include markup/styles unless there is supporting documentation.
- [x] Code does not use `.html_safe`.
- [x] All images added/updated have alt text.
- [x] Does not bypass RuboCop rules in any way.

# PR Type
What kind of change does this PR introduce?:

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix, Migration, or Report (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/188559648

# A brief description of the changes:

Current behavior: Admin can see multiple options when viewing AI/AN verifications on users, mostly being unnecessary.

New behavior: Admin will only see View History as a dropdown option on AI/AN verifications when the feature flag is on

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and indicate which client(s) it applies to.

Variable name: ai_an_self_attestation, AI_AN_SELF_ATTESTATION_IS_ENABLED

- [ ] DC
- [x] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

### CODE REVIEW

**Review by charlienparker — DISMISSED** [review]

File: app/helpers/verification_helper.rb [hunk 1]
```diff
@@ -293,7 +293,10 @@ def ridp_admin_actions(ridp_type, person)
   end
 
   def build_admin_actions_list(v_type, f_member)
-    if f_member.consumer_role.aasm_state == 'unverified' || VerificationType::ADMIN_CALL_HUB_VERIFICATION_TYPES.exclude?(v_type.type_name)
+    if EnrollRegistry.feature_enabled?(:ai_an_self_attestation) && v_type.type_name == VerificationType::AMERICAN_INDIAN_STATUS
+      attested_only = ["Verify", "Reject", "Call HUB", "Extend"]
+      ::VlpDocument::ADMIN_VERIFICATION_ACTIONS.reject{ |action| attested_only.include?(action) }
```
Comment (charlienparker) [inline app/helpers/verification_helper.rb:298]: for readability sake, I'd offer to use the `-` operator instead of `reject`, but super nitpicky

**Review by charlienparker — DISMISSED** [review]

File: app/helpers/verification_helper.rb [hunk 2]
```diff
@@ -293,7 +293,10 @@ def ridp_admin_actions(ridp_type, person)
   end
 
   def build_admin_actions_list(v_type, f_member)
-    if f_member.consumer_role.aasm_state == 'unverified' || VerificationType::ADMIN_CALL_HUB_VERIFICATION_TYPES.exclude?(v_type.type_name)
+    if EnrollRegistry.feature_enabled?(:ai_an_self_attestation) && v_type.type_name == VerificationType::AMERICAN_INDIAN_STATUS
+      attested_only = ["Verify", "Reject", "Call HUB", "Extend"]
```
Comment (mdkaraman) [inline app/helpers/verification_helper.rb:297]: This seems like a good opportunity to break up the values in `::VlpDocument::ADMIN_VERIFICATION_ACTIONS` to each be defined as their own constant, and we can convert `ADMIN_VERIFICATION_ACTIONS` into a combination of those new constants.  If we do that, we can just return something like `::VlpDocument::ViewHistory` here. This way we avoid hardcoding values outside of a class where the definition of what those values are really belongs to that class.

**Review by mdkaraman — COMMENTED** [review]
Nice work on the spec coverage!  Just one refactor I think we should include in this change that I noted.

**Review by mdkaraman — DISMISSED** [review]
Looks good!

**Review by charlienparker — APPROVED** [review]

**Review by mdkaraman — APPROVED** [review]

**Pull request merged**